iosxr_store_box.py

```diff
@@ -63,32 +63,51 @@ def main(argv):
-            print('==> E.g. iosxrv-x64-vbox/iosxr_store_box.py -b iosxrv-fullk9-x64.box -r -v -m "Latest box for release."')
-            sys.exit()
-        if opt in ('-b', '--box'):
-            input_box = arg
-        if opt in ('-m', '--message'):
-            message = arg
-            if not message:
-                message = 'No reason for update specified'
-        elif opt in ('-r', '--release'):
-            artifactory_release = True
-        elif opt in ('-v', '--verbose'):
-            verbose = True
-        elif opt in ('-t', '--test'):
-            test = True
+    # Suck in the input BOX and handle errors
+    parser = argparse.ArgumentParser(
+        formatter_class=RawDescriptionHelpFormatter,
+        description='A tool to upload an image to a maven repo like artifactory ' +
+        'using curl, the image typically being a vagrant virtualbox, but could ' +
+        'be anything.\n' +
+        'User can select snapshot or release, the release images get synced to ' +
+        'devhub.cisco.com - where they are available to customers.\n' +
+        'This tool also sends an email out to an email address or an alias to ' +
+        'inform them of the new image.\n' +
+        'It is designed to be called from other tools, like iosxr_ios2vbox.py.\n\n' +
+        'It will rely on the following environment variables to work:\n ' +
+        'ARTIFACTORY_USERNAME\n ' +
+        'ARTIFACTORY_PASSWORD\n ' +
+        'ARTIFACTORY_LOCATION_SNAPSHOT\n ' +
+        'ARTIFACTORY_LOCATION_RELEASE\n ' +
+        'ARTIFACTORY_SENDER\n ' +
+        'ARTIFACTORY_RECEIVER',
+        epilog="E.g.:\n" +
+        "iosxrv-x64-vbox/iosxr_store_box.py -b iosxrv-fullk9-x64.box --release --verbose --message 'A new box because...'\n" +
+        "iosxrv-x64-vbox/iosxr_store_box.py -b iosxrv-fullk9-x64.box --release, --message 'A new box because...'\n"
+        "iosxrv-x64-vbox/iosxr_store_box.py -b iosxrv-fullk9-x64.box -r -v -m 'Latest box for release.'\n")
+
+    parser.add_argument('BOX_FILE',
+                        help='BOX filename')
+    parser.add_argument('-m', '--message', nargs='?', metavar="'New box reason'",
+                        const='No reason for update specified',
+                        help='Optionally specify a reason for uploading this box.')
+    parser.add_argument('-r', '--release', action='store_true',
+                        help='store in appdevci-release where it will get synced to devhug')
+    parser.add_argument('-v', '--verbose', action='store_true',
+                        help='turn on verbose messages')
+    parser.add_argument('-t', '--test', action='store_true',
+                        help='test only')
+
+    args = parser.parse_args()
+
+    input_box = args.BOX_FILE
+    if args.message is None:
+        # User did not add -m at all - and we always need a message
+        args.message = 'No reason for update specified'
+    message = args.message
```

Comment:
No problem here. I wouldn't do it for a variable you're only going to use once or twice (clearer to just use the original in that case), but if you're wanting to reference it a bunch, this is a fine way to save some characters.